draft-ietf-tls-tls13.md

```diff
@@ -5045,17 +5051,20 @@ in \[].
                                v
                             NEGOTIATED
                                | Send ServerHello
+                               | K_send = handshake
                                | Send EncryptedExtensions
                                | [Send CertificateRequest]
 Can send                       | [Send Certificate + CertificateVerify]
 app data -->                   | Send Finished
-after                 +--------+--------+
-here         No 0-RTT |                 | 0-RTT
-                      |                 v
-                      |             WAIT_EOED <---+
+after                          | K_send = application
+here                  +--------+--------+
+             No 0-RTT |                 | 0-RTT
+   K_recv = handshake |                 | K_recv = early_data
+[Skip decrypt errors] |             WAIT_EOED <---+
                       |            Recv |   |     | Recv
-                      |  EndOfEarlyData |   |     | early data
-                      |                 |   +-----+
+                      |  EndOfEarlyData |   |     | Early data
+                      |        K_recv = |   +-----+
+                      |       handshake |
```

Comment:
I don't know how to improve this, but this is a little cramped.  Maybe...
```
  K_recv = handshake  |                 |                           
[Skip decrypt errors] |    +------> WAIT_EOED --+                    
                      |    |       Recv |       | Recv EndOfEarlyData
                      |    | early data |       | K_recv = handshake
                      |    +------------+       |
                      |                         |
                      +> WAIT_FLIGHT2 <---------+
```
That's 69 columns exactly, though you can trim one extra.